tests/pipelines/stepping.py

```diff
@@ -0,0 +1,148 @@
+"""
+Pipeline stepping is a way to advance the asynchronous data pipeline
+deterministically by exposing each step separately and advancing the states
+manually.
+"""
+
+
+import functools
+import time
+import types
+import logging
+from contextlib import contextmanager
+from unittest.mock import patch
+
+import bigchaindb.core
+from bigchaindb.backend.changefeed import ChangeFeed
+import bigchaindb.pipelines.block
+import bigchaindb.pipelines.stale
+import bigchaindb.pipelines.vote
+
+
+class MultipipesStepper:
```

Comment:
Thanks! :D I did it kinda quickly. I should put a block comment in `MultipipesStepper`, it's basically an analogue of `Pipeline` for deterministic stepping of the pipeline.